### PR TITLE
Update documentation related to running e2e and unit tests

### PR DESCRIPTION
**Pull request by juliaamosova**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Update documentation related to how to run e2e and unit tests. 

### How to test the changes in this Pull Request:

This is just a documentation update so there is no need to test this PR. 

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

N/A

### CODE REVIEW

File: tests/e2e-tests/README.md [hunk 1]
```diff
@@ -7,8 +7,8 @@ Automated end-to-end tests for WooCommerce.
 - [Pre-requisites](#pre-requisites)
   - [Install NodeJS](#install-nodejs)
   - [Install Docker](#install-docker)
-  - [Configuration](#configuration)
-      - [Test Environment Configuration](#test-environment-configuration)
+  - [Configuration (how it works under the hood)](#configuration-(how it-works-under-the-hood))
```
Comment (ObliviousHarmony) [inline tests/e2e-tests/README.md]: This change breaks the markdown!

**Review by ObliviousHarmony — CHANGES_REQUESTED** [review]
The changes look good, just one small thing :)

File: tests/e2e-tests/README.md (hunk 1 above)
Comment (juliaamosova) [reply tests/e2e-tests/README.md]: Thanks for catching it @ObliviousHarmony! I corrected it and added a few more details and explanations to the docs.

**Review by ObliviousHarmony — APPROVED** [review]
Looks great!

**Pull request merged**